### PR TITLE
Prod 393: All time leaderboard

### PR DESCRIPTION
**Pull request by yash431garg**
- Description
All time leaderboard page and tab
- What are the steps to test that this code is working?
A new all time leaderboard tab on leaderbaord page which will navigate to all time leaderboard page
- Screen shots or recordings for UI changes
Screenshot 2024-10-14 at 5 18 17 PM

### CODE REVIEW

**Comment by yash431garg** [comment]
Before merge we need to may be change the icon for all time tab. 

cc: @marvinmarnold @Yihwan

File: app/application/leaderboard/all-time/page.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Leaderboard from "@/app/components/Leaderboard/Leaderboard";
+import { getCurrentUser } from "@/app/queries/user";
+import { redirect } from "next/navigation";
+
+const AllTimeLeaderboard = async () => {
+  const user = await getCurrentUser();
+
+  if (!user) redirect("login");
```
Comment (danijel-radomiljac) [inline app/application/leaderboard/all-time/page.tsx:8]: /login should go

Comment (yash431garg) [reply app/application/leaderboard/all-time/page.tsx]: Just added it in the same pr @danijel-radomiljac

**Comment by gatorthana** [comment]
@yash431garg here you go!

https://www.figma.com/design/3pZWm9gRbVMnkAfd5zY7GE/Chomp-UI-Shared-Library?node-id=2452-357&node-type=canvas&t=XnUaUZH5spt90Y6W-11

File: app/actions/leaderboard.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { getCurrentUser } from "../queries/user";
 import prisma from "../services/prisma";
 import { getStartAndEndOfDay, getWeekStartAndEndDates } from "../utils/date";
 interface LeaderboardProps {
-  variant: "weekly" | "daily" | "stack";
+  variant: "weekly" | "daily" | "stack" | "all-time";
```
[the remark this answers is not in the document]
Comment (yash431garg) [reply app/actions/leaderboard.ts:21]: Sure,

**Comment by marvinmarnold** [comment]
@yash431garg test is failing btw

**Comment by yash431garg** [comment]
> @yash431garg test is failing btw

Yea, it is failing because rpc url is missing in workflow secret keys. I will check if I can mock this otherwise we can add that key.

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Update workflow so that env supplied to test: https://github.com/gator-labs/chomp/blob/5ff2ade4099aec6b5b817380072f9778829f0999/.github/workflows/test.yml#L25

File: __tests__/actions/get_alltime-leaderboard.test.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  getJwtPayload: jest.fn(),
+}));
+jest.mock("next/navigation", () => ({
+  redirect: jest.fn(),
+}));
+jest.mock("next/cache", () => ({
+  revalidatePath: jest.fn(),
+}));
+
+describe("Get All-time leaderboard data", () => {
```
Comment (marvinmarnold) [inline __tests__/actions/get_alltime-leaderboard.test.ts:23]: jest.setTimeout(30_000)

Comment (marvinmarnold) [reply __tests__/actions/get_alltime-leaderboard.test.ts:23]: before describe

**Review by danijel-radomiljac — APPROVED** [review]

**Review by danijel-radomiljac — APPROVED** [review]

**Pull request merged**